tests/unit/parallel/lexicographical_compare.cpp

```diff
@@ -0,0 +1,475 @@
+    test_lexicographical_compare_exception(seq, IteratorTag());
+    test_lexicographical_compare_exception(par, IteratorTag());
+
+    test_lexicographical_compare_async_exception(seq(task), IteratorTag());
+    test_lexicographical_compare_async_exception(par(task), IteratorTag());
+
+    test_lexicographical_compare_exception(execution_policy(par), IteratorTag());
+    test_lexicographical_compare_exception(execution_policy(seq(task)), IteratorTag());
+    test_lexicographical_compare_exception(execution_policy(par(task)), IteratorTag());
+}
+
+void lexicographical_compare_exception_test()
+{
+    test_lexicographical_compare_exception<std::random_access_iterator_tag>();
+    test_lexicographical_compare_exception<std::forward_iterator_tag>();
+}
+
+//////////////////////////////////////////////////////////////////////////////
+template <typename ExPolicy, typename IteratorTag>
+void test_lexicographical_compare_bad_alloc(ExPolicy const& policy, IteratorTag)
+{
+    BOOST_STATIC_ASSERT(hpx::parallel::is_execution_policy<ExPolicy>::value);
+
+    typedef std::vector<std::size_t>::iterator base_iterator;
+    typedef test::decorated_iterator<base_iterator, IteratorTag>
+        decorated_iterator;
+
+    std::vector<std::size_t> c(10007);
+    std::fill(boost::begin(c), boost::end(c), std::rand() + 1);
+
+    std::vector<std::size_t> h(10006);
+    std::fill(boost::begin(h), boost::end(h), std::rand() + 1);
+
+    bool caught_bad_alloc = false;
+    try {
+        hpx::parallel::lexicographical_compare(policy,
+            decorated_iterator(
+                boost::begin(c),
+                [](){ throw std::bad_alloc(); }),
+            decorated_iterator(
+                boost::end(c),
+                [](){ throw std::bad_alloc(); }),
+            boost::begin(h), boost::end(h));
+        HPX_TEST(false);
+    }
+    catch(std::bad_alloc const&) {
+        caught_bad_alloc = true;
+    }
+    catch(...) {
+        HPX_TEST(false);
+    }
+
+    HPX_TEST(caught_bad_alloc);
+}
+
+template <typename ExPolicy, typename IteratorTag>
+void test_lexicographical_compare_async_bad_alloc(ExPolicy const& p, IteratorTag)
+{
+    typedef std::vector<std::size_t>::iterator base_iterator;
+    typedef test::decorated_iterator<base_iterator, IteratorTag>
+        decorated_iterator;
+
+    std::vector<std::size_t> c(10007);
+    std::fill(boost::begin(c), boost::end(c), std::rand() + 1);
+
+    std::vector<std::size_t> h(10006);
+    std::fill(boost::begin(h), boost::end(h), std::rand() + 1);
+
+    bool caught_bad_alloc = false;
+    try {
+        hpx::future<bool> f =
+            hpx::parallel::lexicographical_compare(p,
+                decorated_iterator(
+                    boost::begin(c),
+                    [](){ throw std::bad_alloc(); }),
+                decorated_iterator(
+                    boost::end(c),
+                    [](){ throw std::bad_alloc(); }),
+                boost::begin(h), boost::end(h));
+
```

Comment:
Same as above, please add an additional test to verify that the algorithm itself didn't throw.